AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt

```diff
@@ -579,6 +589,26 @@ abstract class AbstractFlashcardViewer :
         // Reset the activity title
         updateActionBar()
         selectNavigationItem(-1)
+        refreshIfRequired(isResuming = true)
+    }
+
+    @NeedsTest("if opExecuted is called while activity is in the background, audio plays onResume")
+    private fun refreshIfRequired(isResuming: Boolean = false) {
+        // Defer the execution of `opExecuted` until the user is looking at the screen.
+        // This ensures that audio/timers are not accidentally started
+        if (!isResuming && !lifecycle.currentState.isAtLeast(Lifecycle.State.RESUMED)) {
+            if (refreshRequired != null) {
+                Timber.d("deferred refresh as activity was not STARTED")
+            }
+            return
+        }
+        // refresh
+        refreshRequired?.let {
+            Timber.d("refreshIfRequired: redraw")
+            // if changing code, re-evaluate `refreshRequired = null` in `updateCardAndRedraw`
+            launchCatchingTask { updateCardAndRedraw() }
+            refreshRequired = null
+        }
```

Comment:
Just a stab at inversing the boolean, not sure spacing is correct and may have borked the actual conditional test 

```suggestion
        // refresh if we are already resumed or we are resuming
        if (isResuming || lifecycle.currentState.isAtLeast(Lifecycle.State.RESUMED)) {
          refreshRequired?.let {
              Timber.d("refreshIfRequired: redraw")
              // if changing code, re-evaluate `refreshRequired = null` in `updateCardAndRedraw`
              launchCatchingTask { updateCardAndRedraw() }
              refreshRequired = null
          }
        } else {
            if (refreshRequired != null) {
                Timber.d("deferred refresh as activity was not STARTED")
            }
            return
        }
```

Reply:
Sorry! I did the work before seeing this comment. Re-pushed as the following, which I feel is a little neater (no `else { if }` and no `return`

```kotlin
        if (isResuming || lifecycle.currentState.isAtLeast(RESUMED)) {
            refreshRequired?.let {
                Timber.d("refreshIfRequired: redraw")
                // if changing code, re-evaluate `refreshRequired = null` in `updateCardAndRedraw`
                launchCatchingTask { updateCardAndRedraw() }
                refreshRequired = null
            }
        } else if (refreshRequired != null) {
            // onResume() will execute this method
            Timber.d("deferred refresh as activity was not STARTED")
        }
```

Reply:
yeah, that doesn't hurt my head when I read it,  +1